### ISSUE TITLE
syntax error in configuration file causes TypeError exception

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1150]
```
#!python

# Sphinx version: 1.2b1
# Python version: 2.7.2
# Docutils version: 0.10 release
# Jinja2 version: 2.6
Traceback (most recent call last):
  File "C:\Program Files (x86)\SwTools\Python\lib\site-packages\sphinx-1.2b1-py2.7.egg\sphinx\cmdline.py", line 246, in main
    warningiserror, tags, verbosity, parallel)
  File "C:\Program Files (x86)\SwTools\Python\lib\site-packages\sphinx-1.2b1-py2.7.egg\sphinx\application.py", line 137, in __init__
    self._init_env(freshenv)
  File "C:\Program Files (x86)\SwTools\Python\lib\site-packages\sphinx-1.2b1-py2.7.egg\sphinx\application.py", line 182, in _init_env
    return self._init_env(freshenv=True)
  File "C:\Program Files (x86)\SwTools\Python\lib\site-packages\sphinx-1.2b1-py2.7.egg\sphinx\application.py", line 164, in _init_env
    self.env.find_files(self.config)
  File "C:\Program Files (x86)\SwTools\Python\lib\site-packages\sphinx-1.2b1-py2.7.egg\sphinx\environment.py", line 344, in find_files
    self.srcdir, config.source_suffix, exclude_matchers=matchers))
  File "C:\Program Files (x86)\SwTools\Python\lib\site-packages\sphinx-1.2b1-py2.7.egg\sphinx\util\__init__.py", line 85, in get_matching_docs
    suffixpattern = '*' + suffix
TypeError: cannot concatenate 'str' and 'tuple' objects

```

Caused by "source_suffix = '.rst', '.py'" int the conf.py file. I just started trying Sphinx and without having read the documentation tried something to include all *.rst and *.py files. 

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1150
- Originally reported by: [Humbalan](https://bitbucket.org/Humbalan)
- Originally created at: 2013-04-22T23:01:11.142

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-04-23 01:00:17+00:00_

TypeError is correct because conf.py conform to Python syntax.
Please read http://sphinx-doc.org/config.html

However, I think the error message is unkind too.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-13 06:36:18+00:00_

This is actually not trivial: we'd need a system of type-annotation for all config values.

**Comment by shimizukawa** [comment]
_From Jan Ulrich Hasecke on 2014-01-23 10:55:27+00:00_

Would it be possible to type check all config values beforehand and give helpful error messages?

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-23 12:58:07+00:00_

As I said: only if we annotate all the values with their permissible type. That would be a major API change (but probably a good idea nevertheless...)

**Comment by shimizukawa** [comment]
_From Robert Lehmann on 2014-07-29 07:38:05+00:00_

I have a patch ready that _warns_ if the configured value does not share a common baseclass with the default value.  That does not require a “major API change.”

Polishing up and posting in the next few days.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-09-29 04:36:37+00:00_

[Robert Lehmann](https://bitbucket.org/lehmannro) how about the progress?

**Comment by shimizukawa** [comment]
_From Robert Lehmann on 2014-10-31 08:37:35+00:00_

See https://bitbucket.org/birkenfeld/sphinx/pull-request/314.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-12-04 08:22:25+00:00_

Checked configuration values for their types. Fixes #1150.

→ <<cset 5cc72ca52bf42239a4bf1eb8df276545e886198e>>
